File: NotificareSdk.Push.iOS.Binding.Bridge/NotificarePushBinding/Sources/Models/NotificareTransport.swift
import Foundation
import NotificarePushKit

@objc
public enum NotificareTransport: Int {
    case notificare
    case apns
    case unknown

    public init(from transport: NotificarePushKit.NotificareTransport?) {
        switch transport {
        case .notificare:
            self = .notificare
        case .apns:
            self = .apns
        case .none:
            self = .unknown
        @unknown default:
            self = .unknown
        }
    }
}

File: Notificare.Push.Core/Models/NotificareTransport.cs
namespace NotificareSdk.Push.Core.Models;

public enum NotificareTransport
{
    Notificare,
    GCM,
    APNS,
}

public static class NotificareTransportExtensions
{
    public static string ToRawValue(this NotificareTransport transport)
    {
        return transport switch
        {
            NotificareTransport.Notificare => "Notificare",
            NotificareTransport.GCM => "GCM",
            NotificareTransport.APNS => "APNS",
            _ => throw new ArgumentException($"Unknown {nameof(NotificareTransport)}: {transport}")
        };
    }
}
